tests/src/Traits/MailConfigTrait.php

```diff
@@ -34,6 +34,28 @@ protected function checkMailConfigOverride(): void {
     }
   }
 
+  /**
+   * Clears the config values from both cache and $_GLOBALS array.
+   *
+   * This ensures that when values are overridden or changed in the settings.php
+   * file during the test run, the values are being read anew from storage.
+   * Config factory sets overrides both in cache and in the $_GLOBALS array so
+   * cleaning them would require to remove them from both indexes.
+   *
+   * When we are changing the settings values before running behat tests, the
+   * values are already loaded into cache and $_GLOBALS so checking for
+   * overrides will still fail.
```

Comment:
If this is needed because the settings are being changed, then it's better to call this code at the time the settings change, instead of at the time the settings are read. Otherwise this will be a problem every time we read from the settings file and forget to call this method.